### PR TITLE
space traveler development

### PR DESCRIPTION
**Pull request by Joel-100**
## All Completed tasks

- create App and setup dependencies
- Add header to a website with a Navbar

- We set up the mission
- We fetched the data from API 
- We displayed the mission on our mission page
- We implemented mission joining
- We implemented mission leave
- We switched badges for missions
- We implemented missions on the profile page

- We set up the rockets
- We fetched the data from API 
- We displayed the mission on our rocket page
- We implemented rocket joining
- We implemented rocket reserve
- We switched badges for rockets
- We implemented rockets on the profile page

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @Joel-100, @freddykaberuka, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a professional README however microverse changed the template, kindly consider using [the new README template provided by microverse](https://github.com/microverseinc/readme-template)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 1]
```diff
@@ -1 +1,59 @@
...
+- Twitter: [@joelkugonza](https://twitter.com/JoelKugonza)
+- LinkedIn: [@joelkugonza](https://www.linkedin.com/in/joel-kugonza-879211190/)
+
+👤 **Freddy Kaberuka**
+
+- GitHub: [@freddykaberuka](https://github.com/freddykaberuka)
+- Twitter: [@trojanx07](https://twitter.com/trojanx07)
+- LinkedIn: [@freddykaberuka](https://www.linkedin.com/in/kaberuka-freddy-853b08153/)
+
+
```
Comment (ElsonOtake) [inline README.md:40]: - [ ] Kindly add the "future features" session to your README. Makes the information provided on your README more professional. You can use the following [template](https://github.com/microverseinc/readme-template).

File: src/App.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
```
Comment (ElsonOtake) [inline src/App.test.js:3]: - [ ] Please create tests for all your components and functions. The tests are important tools for the knowledge of developers' careers.

Screen Shot 2022-11-19 at 2 07 39 PM

File: src/pages/Rockets.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+function Rockets() {
+  const rockets = useSelector((state) => state.rockets);
+  const dispatch = useDispatch();
+
+  const reserveRocket = (id) => {
+    dispatch(toggleReservation(id));
+  };
+
+  useEffect(() => {
+    dispatch(fetchAllRockets());
```
Comment (ElsonOtake) [inline src/pages/Rockets.js:15]: - [ ] I think it's better to dispatch the action just once. Will create a more professional code and fix the issue of resetting all your reservations when reloading the rockets list.

File: src/redux/missions/mission.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export const FetchMissionsHandler = () => async (dispatch) => {
+  const response = await axios.get(url);
+  return dispatch({ type: FETCH_MISSIONS, payload: response.data });
+};
+
+export const ToggleMission = (id) => ({ type: TOGGLE_MISSION, id });
+
+const missionsReducer = (state = missions, action) => {
+  switch (action.type) {
+    case FETCH_MISSIONS:
```
Comment (ElsonOtake) [inline src/redux/missions/mission.js:17]: - [ ] Please store just the required data from the API. It's a way to reduce the volume of area needed to store the data and makes your app run lightly.

Screen Shot 2022-11-19 at 2 38 10 PM

File: src/pages/Rockets.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  }, []);
+  return (
+    <div className="rocket-container">
+      {rockets.map((item) => (
+        <div className="rocket-card" key={item.id}>
+          <div className="rocket-profile">
+            <img src={item.flickr_images} alt="" />
```
Comment (ElsonOtake) [inline src/pages/Rockets.js:22]: - [ ] I think it's better to use the same width for all your images. Will create a better UI.

Screen Shot 2022-11-19 at 12 58 33 PM

File: src/pages/Mission.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          </tr>
+        </thead>
+        <tbody>
+          {missions.map((el) => (
+            <tr key={el.mission_id}>
+              <td className="missionName">{el.mission_name}</td>
+              <td className="missionDescription">{el.description}</td>
+              <td className="btnRow">
+                <span className={el.active ? 'activeMember' : 'inactiveMember'}>
+                  {el.active ? 'Active Member' : 'Not A member'}
```
Comment (ElsonOtake) [inline src/pages/Mission.js:32]: - [ ] Please fix the issue with the text overlapping for some sizes of screens. Will create a better UI.

Screen Shot 2022-11-19 at 2 46 40 PM

**Review by ElsonOtake — CHANGES_REQUESTED** [review]
STATUS: CHANGES REQUIRED 🟡 

Hi @Joel-100 @freddykaberuka,

Good job so far! 🎉 🎉 
There are some issues that you still need to work on to go to the next project but you are almost there! 🥲 🥲 

woohoo-youre-almost

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/Rockets.js (hunk 5 above)
Comment (Joel-100) [reply src/pages/Rockets.js:22]: - The project doesn't require us to have responsiveness

File: src/pages/Mission.js (hunk 6 above)
Comment (Joel-100) [reply src/pages/Mission.js:32]: The project doesn't require us to have responsiveness

File: src/components.test.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { render } from '@testing-library/react';
```
Comment (Rustamxon7) [inline src/components.test.js:1]: You did a great job implementing the changes given by the previous reviewer. However, you have some missing parts and some errors that you need to handle before approving this project. So, let's get started :)

- [ ] 1. When I run `npm test` I get errors after a few seconds: 
 Screenshot 2022-11-20 at 00 16 32
 Kindly fix them :)

File: src/components.test.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import { Provider } from 'react-redux';
+import store from './redux/configStore';
+import Rockets from './pages/Rockets';
+import Mission from './pages/Mission';
+import Profile from './pages/Profile';
+
+describe('Test UI', () => {
```
Comment (Rustamxon7) [inline src/components.test.js:8]: - [ ]  2. Based on the previous reviewer's request and the requirement you must write tests for components and functions for your project. You did good by adding a snapshot test but you also need to write tests for functions as well.  Kindly add them, please. If you need more information you can slack me :)

**Review by Rustamxon7 — CHANGES_REQUESTED** [review]
Hello @Joel-100,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components.test.js (hunk 7 above)
Comment (Joel-100) [reply src/components.test.js:1]: The tests are passing on my end. See screenshot below

image

Comment (Rustamxon7) [reply src/components.test.js:1]: Yes, but wait for it. It's showing up after the test complete.

**Review by vickymarz — APPROVED** [review]
Hi @Joel-100, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- All tests passed ✅ 
- Linters are working properly ✅ 


## Optional suggestions

- I will strongly recommend you create a folder to contain all your test files so that your codebase can be more organized 👍🏽 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**